gym-unity/tests/test_gym.py

```diff
@@ -44,3 +45,27 @@ def test_multi_agent(mock_communicator, mock_launcher):
     assert isinstance(rew, list)
     assert isinstance(done, list)
     assert isinstance(info, dict)
+
+@mock.patch('gym_unity.envs.unity_env.UnityEnvironment')
+def test_branched_flatten(mock_env):
+    mock_env.return_value.academy_name = 'MockAcademy'
+    mock_brain = mock.Mock();
+    mock_brain.return_value.number_visual_observations = 0
+    mock_brain.return_value.num_stacked_vector_observations = 1
+    mock_brain.return_value.vector_action_space_type = 'discrete'
+    mock_brain.return_value.vector_observation_space_size = 1
+    # Unflattened action space
+    mock_brain.return_value.vector_action_space_size = [2,2,3]
+
+    mock_env.return_value.brains = {'MockBrain':mock_brain()}
+    mock_env.return_value.external_brain_names = ['MockBrain']
+    env = UnityEnv(' ', use_visual=False, multiagent=False, flatten_branched=True)
+    assert isinstance(env.action_space, spaces.Discrete)
+    assert env.action_space.n==12
+    assert env._flattener.lookup_action(0)==[0,0,0]
+    assert env._flattener.lookup_action(11)==[1,1,2]
+
+    # Check that False produces a MultiDiscrete
+    env = UnityEnv(' ', use_visual=False, multiagent=False, flatten_branched=False)
+    assert isinstance(env.action_space, spaces.MultiDiscrete)
+    
```

Comment:
Seems to be an erroneous EOF character (?)

Reply:
Don't all of our Python files have an extra newline at the end? Or is there another character I can't see?

Reply:
@ervteng - one thing to consider is to add a "macro" to your IDE to remove all trailing whitespace and ensure a newline character (this can be triggered on-save).